### DISCUSSION TITLE
How can I get all wallets without providing projectId?

### DISCUSSION THREAD

**Discussion by wisdomabioye** [rainbow-me/rainbowkit#1398] [Q&A] [answered]
I don't want to use Wallet Connect (I'm not providing the projectId), I need to get all `wallets` and pass it to `connectorsForWallets`

```
const { connectors, wallets } = getDefaultWallets({
    appName: AppInfo.name,
    projected: '', // not using this
    chains
})

connectors = connectorsForWallets(wallets);
```

**Comment by DanielSinclair** [comment, marked as answer]
This is not directly supported, as most of the RainbowKit wallet connectors support both the WalletConnect and Injected Provider paths (depending on the platform or if a wallet is detected). However, you can use our [Custom Wallet API](https://www.rainbowkit.com/docs/custom-wallets) to implement your own wallets/forks of wallet. You can also reference the list of [Supported Wallets](https://www.rainbowkit.com/docs/custom-wallet-list#specific-wallets) to see which connectors require a `projectId`, and opt to avoid adding those to your wallet list. An example implementation may look like this:
```tsx
import { injectedWallet } from '@rainbow-me/rainbowkit/wallets';
const connectors = connectorsForWallets([
  {
    groupName: 'Recommended',
    wallets: [
      injectedWallet({ chains }),
    ],
  },
]);
```
But it is best practice to support WalletConnect because wallets rely on this protocol for a wide range of uses, including smart contract wallet support. Getting a `projectId` should only take a few minutes, and is free.
